ui/src/app/views/workspace-edit/component.tsx

```diff
@@ -0,0 +1,584 @@
+      </TooltipTrigger>
+      }
+    </div>
+    {props.subHeader && <div style={{...styles.header, color: '#4A4A4A', fontSize: 14}}>
+      {props.subHeader}
+    </div>
+    }
+    <div style={styles.text}>
+      {props.text}
+    </div>
+    <div>
+      {props.children}
+    </div>
+  </div>;
+};
+
+export const WorkspaceCategory = (props) => {
+  return <div style={...fp.merge(styles.categoryRow, props.style)}>
+    <input style={{height: '.66667rem', marginRight: '.31667rem'}} type='checkbox'
+           checked={!!props.value}
+           onChange={props.onChange ? (e => props.onChange(e.target.checked)) : undefined}/>
+    <div style={{display: 'flex', flexDirection: 'column', marginTop: '-0.2rem'}}>
+      <label style={styles.shortDescription}>
+        {props.item.shortDescription}
+      </label>
+      <div>
+        <label style={styles.longDescription}>
+          {props.item.longDescription}
+        </label>
+        {props.children}
+      </div>
+    </div>
+  </div>;
+};
+
+export enum WorkspaceEditMode { Create = 1, Edit = 2, Clone = 3 }
+
+
+export interface WorkspaceEditProps {
+  routeConfigData: any;
+  workspace: Workspace;
+}
+
+export interface WorkspaceEditState {
+  cdrVersionItems: Array<CdrVersion>;
+  disableButton: boolean;
+  workspace: Workspace;
+  workspaceCreationConflictError: boolean;
+  workspaceCreationError: boolean;
+  cloneUserRole: boolean;
+  loading: boolean;
+}
+
+export const WorkspaceEdit = withRouteConfigData()(withCurrentWorkspace()(
+  class extends React.Component<WorkspaceEditProps, WorkspaceEditState> {
+
+    constructor(props: WorkspaceEditProps) {
+      super(props);
+      this.isEmpty.bind(this);
+      this.state = {
+        cdrVersionItems: [],
+        disableButton: true,
+        workspace: {
+          name: '',
+          description: '',
+          dataAccessLevel: DataAccessLevel.Registered,
+          cdrVersionId: '',
+          researchPurpose: {
+            diseaseFocusedResearch: false,
+            methodsDevelopment: false,
+            controlSet: false,
+            aggregateAnalysis: false,
+            ancestry: false,
+            commercialPurpose: false,
+            population: false,
+            reviewRequested: false,
+            containsUnderservedPopulation: false,
+            underservedPopulationDetails: []
+          }
+        },
+        workspaceCreationConflictError: false,
+        workspaceCreationError: false,
+        cloneUserRole: false,
+        loading: false
+      };
+    }
+
+    componentWillMount() {
+      this.updateWorkspace();
+      this.setCdrVersions();
+    }
+
+
+    updateWorkspace() {
+      if (this.isMode(WorkspaceEditMode.Create)) {
+        const profile =  userProfileStore.getValue().profile;
+        this.setState(fp.set(['workspace', 'namespace'], profile.freeTierBillingProjectName));
+        return;
+      }
+      this.setState({workspace : this.props.workspace});
```

Comment:
Rather than setting this after the component mounts, we could merge this logic into the constructor so we just initialize the state to the right values (remember that we have access to the props in the constructor as well).